homeassistant/components/scrape/__init__.py

```diff
@@ -75,7 +81,7 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
             raise PlatformNotReady
 
         for sensor_config in sensors:
-            discovery.load_platform(
+            await discovery.async_load_platform(
```

Comment:
Can we use `asyncio.gather` for this work instead of looping and awaiting inside the loop?

Reply:
I am a little bit more confused about this one...
The recommendation is:
> Do not await this inside a setup method to avoid a dead lock. Use hass.async_create_task(async_load_platform(..)) instead.

That means we shouldn't use `await`, but surely that means that we shouldn't use `asyncio.gather` either...
Does that mean that the implementation in rest is incorrect?
```python
    if load_tasks:
        await asyncio.gather(*load_tasks)
```

Should we instead be using something like:
```python
    for task in load_tasks:
        hass.async_create_task(task)
```

Reply:
I was trying to search for the quote but couldn't find it. Think also I have seen it from before. which address?

Besides that I think it would make sense to gather the load_tasks and then execute them so will change to that.

Reply:
Let's align with REST for now, and we can resolve `Do not await this inside a setup method to avoid a dead lock.` in a follow-up if it is needed.

Reply:
See https://github.com/home-assistant/core/pull/81291